src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Calendar/Calendar.shared.cs

```diff
@@ -0,0 +1,564 @@
+		/// </summary>
+		public static readonly BindableProperty FirstDayOfWeekProperty =
+			BindableProperty.Create(
+				propertyName: nameof(WeekDayHeaderControlTemplateProperty),
+				returnType: typeof(DayOfWeek),
+				declaringType: typeof(Calendar),
+				defaultValue: DayOfWeek.Monday,
+				propertyChanged: OnFirstDayOfWeekChanged);
+
+		/// <summary>
+		/// Gets or sets first day of week.
+		/// </summary>
+		public DayOfWeek FirstDayOfWeek
+		{
+			get => (DayOfWeek)GetValue(FirstDayOfWeekProperty);
+			set => SetValue(FirstDayOfWeekProperty, value);
+		}
+
+		static void OnFirstDayOfWeekChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateWeekDayHeaders();
+			calendar.UpdateCalendarDays();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="WeekDayHeaderControlTemplate"/> property.
+		/// </summary>
+		public static readonly BindableProperty WeekDayHeaderControlTemplateProperty =
+			BindableProperty.Create(nameof(WeekDayHeaderControlTemplateProperty), typeof(ControlTemplate),
+				typeof(Calendar), propertyChanged: OnWeekDayHeaderControlTemplateChanged);
+
+		/// <summary>
+		/// Gets or sets <see cref="ControlTemplate"/> for week day header at the top of <see cref="Calendar"/>.
+		/// </summary>
+		public ControlTemplate? WeekDayHeaderControlTemplate
+		{
+			get => (ControlTemplate?)GetValue(WeekDayHeaderControlTemplateProperty);
+			set => SetValue(WeekDayHeaderControlTemplateProperty, value);
+		}
+
+		static void OnWeekDayHeaderControlTemplateChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateWeekDayHeaders();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="DayControlTemplate"/> property.
+		/// </summary>
+		public static readonly BindableProperty DayControlTemplateProperty =
+			BindableProperty.Create(nameof(DayControlTemplate), typeof(ControlTemplate), typeof(Calendar),
+				propertyChanged: OnDayControlTemplateChanged);
+
+		/// <summary>
+		/// Gets or sets <see cref="ControlTemplate"/> for <see cref="CalendarDay"/>.
+		/// </summary>
+		public ControlTemplate? DayControlTemplate
+		{
+			get => (ControlTemplate?)GetValue(DayControlTemplateProperty);
+			set => SetValue(DayControlTemplateProperty, value);
+		}
+
+		static void OnDayControlTemplateChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateCalendarDays();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="SelectedDays"/> property.
+		/// </summary>
+		public static readonly BindableProperty SelectedDaysProperty =
+			BindableProperty.Create(nameof(SelectedDays), typeof(IReadOnlyCollection<DateTime>), typeof(Calendar), Enumerable.Empty<DateTime>().ToList(), propertyChanged: OnSelectedDaysChanged);
```

Comment:
Just curious - how come we're using `DateTime` instead of `DateTimeOffset`?

I'd prefer to use `DateTimeOffset` in order to support timezones.

Let me know if there is context that I'm missing!

Reply:
If we move forward with `DateTimeOffset`, we'd need to change all `DateTime` to `DateTimeOffset` in this PR.

Reply:
I've changed DateTime to DateTimeOffset